src/charm.py

```diff
@@ -1231,39 +1231,62 @@ def _get_openstack_runner_manager(
                 name. If None the path in charm state is used.
 
         Returns:
-            An instance of OpenstackRunnerManager.
+            An instance of RunnerScaler.
         """
         if token is None:
             token = state.charm_config.token
         if path is None:
             path = state.charm_config.path
 
-        # Empty image can be passed down due to a delete only case where deletion of runners do not
-        # depend on the image ID being available. Make sure that the charm goes to blocked status
-        # in hook where a runner may be created. TODO: 2024-07-09 This logic is subject to
-        # refactoring.
+        clouds = list(state.charm_config.openstack_clouds_yaml["clouds"].keys())
+        if len(clouds) > 1:
+            logger.warning(
+                "Multiple clouds defined in clouds.yaml. Using the first one to connect."
```

Comment:
Minor nit: Should we print the first cloud name?

Reply:
Can this be done as part of the replacing the openstack clouds yaml with individual openstack configurations?
Since the openstack-clouds-yaml configuration is getting removed, and multiple clouds will not be possible anymore.